.travis.yml

```diff
@@ -11,22 +11,30 @@ matrix:
   allow_failures:
     - env: SHELL=zsh
 script:
-  - >
-    if ${LINT:-false}; then
-      shellcheck -s sh bin/shpec  # Just use th `sh` dialect
-    else
-      time $SHELL bin/shpec shpec/shpec_shpec.sh
-    fi
+  - |
+    ${LINT:-false} &&
+      exec shellcheck -s sh bin/shpec  # `sh` refers to POSIX
+    $SHELL bin/shpec shpec/shpec_shpec.sh
 cache: apt
 before_install:
-  - >
+  - |
+    get_shellcheck() {
+      (
```

Comment:
On 05/03/15 06:13, locochris wrote:
> In .travis.yml <https://github.com/rylnd/shpec/pull/79#discussion_r29552855>:
>
>>  cache: apt
>>  before_install:
>> -  - >
>> +  - |
>> +    get_shellcheck() {
>> +      (
>
> @hlangeveld <https://github.com/hlangeveld> - out of interest, why the subshell?

Clean coding. I'm using variables inside that function.  I even reuse the `$pkg` variable
name, and I try not to pass data through globals.
In ksh, I tend to use `function` and `typeset`, in bash that would be `...()` and `local`.
POSIX does not specify any other way to limit the scope of variables.

David Korn did mention he was adding some Bash features (like `local`) to ksh93,
but that still doesn't make it POSIX.

By the end of the epoch, POSIX shell will have local variables. ;-)
Only 23 more years.